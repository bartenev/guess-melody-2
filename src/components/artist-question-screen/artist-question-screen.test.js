import React from "react";
import renderer from "react-test-renderer";
import ArtistQuestionScreen from "./artist-question-screen";

const question = {
  type: `artist`,
  song: {
    artist: `Jim Beam`,
    src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`,
  },
  answers: [
    {
      picture: `https://lh3.googleusercontent.com/proxy/ELVSk8lMwUNadyXt60N3ckhwjkObTuF7QRCCk5SkhsdUfw7aUO_tFB2XEz99xPS5wePQ9Ey3H6aYGyCAScZbO_1vm6igY_WmCoPkojjNIbdLWOYgbNRPV1xY2h5a5dYL2AWZkt1NM57BkGnBixfw89nTmJSEFsvQvdfgVB-ug4incD1Qe3NUj74`,
      artist: `John Snow`,
    },
    {
      picture: `https://lh3.googleusercontent.com/proxy/ELVSk8lMwUNadyXt60N3ckhwjkObTuF7QRCCk5SkhsdUfw7aUO_tFB2XEz99xPS5wePQ9Ey3H6aYGyCAScZbO_1vm6igY_WmCoPkojjNIbdLWOYgbNRPV1xY2h5a5dYL2AWZkt1NM57BkGnBixfw89nTmJSEFsvQvdfgVB-ug4incD1Qe3NUj74`,
      artist: `Jack Daniels`,
    },
    {
      picture: `https://lh3.googleusercontent.com/proxy/ELVSk8lMwUNadyXt60N3ckhwjkObTuF7QRCCk5SkhsdUfw7aUO_tFB2XEz99xPS5wePQ9Ey3H6aYGyCAScZbO_1vm6igY_WmCoPkojjNIbdLWOYgbNRPV1xY2h5a5dYL2AWZkt1NM57BkGnBixfw89nTmJSEFsvQvdfgVB-ug4incD1Qe3NUj74`,
      artist: `Jim Beam`,
    },
  ]
};

it(`ArtistQuestionScreen is rendered correctly`, () => {
  const tree = renderer.create(
      <ArtistQuestionScreen
        onAnswer={() => {}}
        question={question}
      />, {
        createNodeMock: () => {
          return {};
        }
      }
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

