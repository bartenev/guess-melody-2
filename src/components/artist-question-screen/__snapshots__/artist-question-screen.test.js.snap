// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistQuestionScreen is rendered correctly 1`] = `
<section
  className="game__screen"
>
  <h2
    className="game__title"
  >
    Кто исполняет эту песню?
  </h2>
  <div
    className="game__track"
  >
    <div
      className="track"
    />
  </div>
  <form
    className="game__artist"
  >
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="answer-0"
        name="answer"
        onChange={[Function]}
        type="radio"
        value="artist-0"
      />
      <label
        className="artist__name"
        htmlFor="answer-0"
      >
        <img
          alt="John Snow"
          className="artist__picture"
          src="https://lh3.googleusercontent.com/proxy/ELVSk8lMwUNadyXt60N3ckhwjkObTuF7QRCCk5SkhsdUfw7aUO_tFB2XEz99xPS5wePQ9Ey3H6aYGyCAScZbO_1vm6igY_WmCoPkojjNIbdLWOYgbNRPV1xY2h5a5dYL2AWZkt1NM57BkGnBixfw89nTmJSEFsvQvdfgVB-ug4incD1Qe3NUj74"
        />
        John Snow
      </label>
    </div>
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="answer-1"
        name="answer"
        onChange={[Function]}
        type="radio"
        value="artist-1"
      />
      <label
        className="artist__name"
        htmlFor="answer-1"
      >
        <img
          alt="Jack Daniels"
          className="artist__picture"
          src="https://lh3.googleusercontent.com/proxy/ELVSk8lMwUNadyXt60N3ckhwjkObTuF7QRCCk5SkhsdUfw7aUO_tFB2XEz99xPS5wePQ9Ey3H6aYGyCAScZbO_1vm6igY_WmCoPkojjNIbdLWOYgbNRPV1xY2h5a5dYL2AWZkt1NM57BkGnBixfw89nTmJSEFsvQvdfgVB-ug4incD1Qe3NUj74"
        />
        Jack Daniels
      </label>
    </div>
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="answer-2"
        name="answer"
        onChange={[Function]}
        type="radio"
        value="artist-2"
      />
      <label
        className="artist__name"
        htmlFor="answer-2"
      >
        <img
          alt="Jim Beam"
          className="artist__picture"
          src="https://lh3.googleusercontent.com/proxy/ELVSk8lMwUNadyXt60N3ckhwjkObTuF7QRCCk5SkhsdUfw7aUO_tFB2XEz99xPS5wePQ9Ey3H6aYGyCAScZbO_1vm6igY_WmCoPkojjNIbdLWOYgbNRPV1xY2h5a5dYL2AWZkt1NM57BkGnBixfw89nTmJSEFsvQvdfgVB-ug4incD1Qe3NUj74"
        />
        Jim Beam
      </label>
    </div>
  </form>
</section>
`;
